packages/svelte/src/internal/client/dom/blocks/each.js

```diff
@@ -258,6 +266,8 @@ function reconcile_indexed_array(array, state, anchor, render_fn, flags) {
 		value = array[i];
 		item = a_items[i];
 		b_items[i] = item;
+		// Indexed each blocks use the value as the key
+		item.k = value;
```

Comment:
surely the index should be the key?

```suggestion
		// Indexed each blocks use the index as the key
		item.k = i;
```

Reply:
Hold on, I just tried again and it works now. Doh. Will refactor